src/main/java/org/spongepowered/mod/mixin/block/data/MixinTileEntity.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.mod.mixin.block.data;
+
+import net.minecraft.util.BlockPos;
+
+import org.spongepowered.api.block.data.TileEntity;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.api.block.BlockLoc;
+import org.spongepowered.api.util.annotation.NonnullByDefault;
+
+@NonnullByDefault
+@Mixin(net.minecraft.tileentity.TileEntity.class)
+public abstract class MixinTileEntity implements TileEntity {
+
+    @Shadow
+    public abstract BlockPos getPos();
+
+    @Override
+    public BlockLoc getBlock() {
+        return (BlockLoc) getPos();
```

Comment:
Via the [`BlockWrapper`](https://github.com/SpongePowered/Sponge/blob/master/src/main/java/org/spongepowered/mod/wrapper/BlockWrapper.java) class.